homeassistant/components/fan/wemo.py

```diff
@@ -118,24 +118,23 @@ def service_handle(service):
         """Handle the WeMo humidifier services."""
         entity_ids = service.data.get(ATTR_ENTITY_ID)
 
+        humidifiers = []
+
+        if entity_ids:
+            humidifiers.extend([device for device in
+                                hass.data[DATA_KEY].values() if
+                                device.entity_id in entity_ids])
+
         if service.service == SERVICE_SET_HUMIDITY:
             target_humidity = service.data.get(ATTR_TARGET_HUMIDITY)
 
-            if entity_ids:
-                humidifiers = [device for device in
-                               hass.data[DATA_KEY].values() if
-                               device.entity_id in entity_ids]
-            else:
-                humidifiers = hass.data[DATA_KEY].values()
+            if not humidifiers:
```

Comment:
Our new standard is that entity_id is required. I think we should change the logic here now or later to follow that. It will be a breaking change.

Reply:
That's funny, I think I had a reviewer ask me to make this optional in the PR where I added the service. I'll go ahead and update it to be required and update the documentation accordingly as well.

Reply:
That was me. We changed this recently.

Reply:
Fair enough. :-) I like them being required better. It's safer that way (no accidental running of services on entities you didn't mean to).